src/deterministic_vi_image_psf/fsm_matrices.jl

```diff
@@ -0,0 +1,175 @@
+
+    fsms.fs0m_conv = zero_sensitive_float_array(
+        StarPosParams, Float64, 1, h_width, w_width);
+
+    # The amount of padding introduced by the convolution
+    (fft_size1, fft_size2) =
+        (h_width + psf_size[1] - 1, w_width + psf_size[2] - 1)
+    # Make sure that the PSF has an odd dimension.
+    @assert psf_size[1] % 2 == 1
+    @assert psf_size[2] % 2 == 1
+    fsms.pad_pix_h = Integer((psf_size[1] - 1) / 2)
+    fsms.pad_pix_w = Integer((psf_size[2] - 1) / 2)
+
+    fsms.fs1m_image_padded = zero_sensitive_float_array(
+        GalaxyPosParams, Float64, 1, fft_size1, fft_size2);
+    fsms.fs1m_conv_padded = zero_sensitive_float_array(
+        GalaxyPosParams, Float64, 1, fft_size1, fft_size2);
+
+    # Brightness images
+    fsms.E_G = zero_sensitive_float_array(
+        CanonicalParams, Float64, num_active_sources, h_width, w_width);
+    fsms.var_G = zero_sensitive_float_array(
+        CanonicalParams, Float64, num_active_sources, h_width, w_width);
+
+    # Store the psf image and its FFT.
+    S = size(psf_image_mat, 1)
+    fsms.psf_fft_vec = Array(Matrix{Complex{Float64}}, S)
+    fsms.psf_vec = Array(Matrix{Float64}, S)
+    for s in 1:size(psf_image_mat, 1)
+        fsms.psf_fft_vec[s] =
+            zeros(Complex{Float64}, fft_size1, fft_size2);
+        fsms.psf_fft_vec[s][1:psf_size[1], 1:psf_size[2]] =
+            psf_image_mat[s, b];
+        fft!(fsms.psf_fft_vec[s]);
+        fsms.psf_vec[s] = psf_image_mat[s, b]
+    end
+end
+
+
+function initialize_fsm_sf_matrices!(
+    fsm_vec::Vector{FSMSensitiveFloatMatrices},
+    ea::ElboArgs{Float64},
+    psf_image_mat::Matrix{Matrix{Float64}})
+
+    # Get the extreme active pixels in each band.
+    h_lower_vec = Int[typemax(Int) for b in ea.images ]
+    w_lower_vec = Int[typemax(Int) for b in ea.images ]
+    h_upper_vec = Int[0 for b in ea.images ]
+    w_upper_vec = Int[0 for b in ea.images ]
+
+    # Since we are using the same size fsm matrix for each source,
+    # and even non-active sources need to be rendered, we need to make
+    #.the fsm matrices as big as the total number of active pixels.
+    for s in 1:ea.S, n in 1:ea.N
+        p = ea.patches[s, n]
+        h1 = p.bitmap_corner[1] + 1
+        w1 = p.bitmap_corner[2] + 1
+        h2 = h1 + size(p.active_pixel_bitmap, 1) - 1
+        w2 = w1 + size(p.active_pixel_bitmap, 2) - 1
+        h_lower_vec[n] = min(h_lower_vec[n], h1)
+        h_upper_vec[n] = max(h_upper_vec[n], h2)
+        w_lower_vec[n] = min(w_lower_vec[n], w1)
+        w_upper_vec[n] = max(w_upper_vec[n], w2)
+    end
+
+    num_active_sources = length(ea.active_sources)
+
+    # Pre-allocate arrays.
+    for b in 1:ea.N
+        initialize_fsm_sf_matrices_band!(
+            fsm_vec[b], b, num_active_sources,
+            h_lower_vec[b], w_lower_vec[b],
+            h_upper_vec[b], w_upper_vec[b],
+            psf_image_mat)
+    end
+end
+
+
+# This is useful for debugging and exploration.  Do not delete.
```

Comment:
Please give this function a name like `debug_populate_fsm_vec!` so it isn't confused with the real `populate_fsm_vec!`. Also please give a date for deleting this in the comment, e.g `Do not delete before 2/2017`, so that we're not maintaining unreachable code indefinitely